app/src/main/java/ch/epfl/sweng/erpa/activities/LoginActivity.java

```diff
@@ -0,0 +1,84 @@
+package ch.epfl.sweng.erpa.activities;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Application;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.graphics.Color;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.annimon.stream.Optional;
+
+import javax.inject.Inject;
+
+import ch.epfl.sweng.erpa.model.UserAuth;
+import ch.epfl.sweng.erpa.R;
+import ch.epfl.sweng.erpa.services.UserAuthService;
+import toothpick.Scope;
+import toothpick.Toothpick;
+
+public class LoginActivity extends Activity {
+    @Inject UserAuthService uas;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);  Application application = getApplication();
+        Scope scope = Toothpick.openScopes(application, this);
+        Toothpick.inject(this, scope);
+        setContentView(R.layout.activity_login);
+    }
+
+    public void login(View view) {
+        String usernameText = ((EditText) findViewById(R.id.username)).getText().toString();
+        String passwordText = ((EditText) findViewById(R.id.password)).getText().toString();
+
+        if (usernameText.isEmpty()) {
+            createPopup(getString(R.string.noNameMessage));
+            return;
+        }
+
+        if (passwordText.isEmpty()) {
+            createPopup(getString(R.string.noPassMessage));
+            return;
+        }
+
+        Optional<UserAuth> ua = uas.getUserAuth(usernameText, passwordText);
+
+        ua.ifPresent(u -> finish());
+
+        createPopup(getString(R.string.incorrectLogin));
+    }
+
+    public void continueWithoutLogin(View view) {
+        Intent intent = new Intent(this, MainActivity.class);
+        startActivity(intent);
+    }
+
+    private void createPopup(String text) {
```

Comment:
Extract this function into an util function, since I believe it's already been implemented by some of your workmates ;-)